src/OpenLoco/Paint/Paint.cpp

```diff
@@ -41,10 +42,128 @@ namespace OpenLoco::Paint
         return &_session;
     }
 
+    void registerHooks()
+    {
+        registerHook(
+            0x004622A2,
+            [](registers& regs) -> uint8_t {
+                registers backup = regs;
+
+                PaintSession session;
+                session.generate();
```

Comment:
It's kinda weird to see a local variable created for this, but it makes sense since it doesn't have any actual local properties. And it's temporary, so 🤷

Reply:
Yeah it's only temporary for the hook